packages/trader/src/Modules/Trading/Components/Form/TradeParams/Duration/__tests__/advanced-duration.spec.tsx

```diff
@@ -0,0 +1,187 @@
+                },
+            ],
+            expiry_type: 'duration',
+            getDurationFromUnit: jest.fn(),
+            number_input_props: {
+                type: 'number',
+                is_incrementable: false,
+            },
+            onChange: jest.fn(),
+            onChangeUiStore: jest.fn(),
+            server_time: 0,
+            shared_input_props: {
+                is_hj_whitelisted: true,
+                max_value: 86400,
+                min_value: 15,
+                onChange: jest.fn(),
+            },
+            start_date: 0,
+        };
+    });
+    const renderAdvancedDuration = (
+        mock_store: ReturnType<typeof mockStore>,
+        default_props: React.ComponentProps<typeof AdvancedDuration>
+    ) => {
+        return render(
+            <TraderProviders store={mock_store}>
+                <AdvancedDuration {...default_props} />
+            </TraderProviders>
+        );
+    };
+    it('Should render mocked button toggle if expiry_list is of length > 1', () => {
+        renderAdvancedDuration(mock_store, default_props);
+        expect(screen.getByText(button_toggle)).toBeInTheDocument();
+    });
+    it('Should not render mocked button toggle if expiry_list is of length <= 1', () => {
+        default_props.expiry_list = [
+            {
+                text: 'Duration',
+                value: 'duration',
+            },
+        ];
+        renderAdvancedDuration(mock_store, default_props);
+        expect(screen.queryByText(button_toggle)).not.toBeInTheDocument();
+    });
+    it('Should render mocked trading date and trading time picker if contract is 24 hours and expiry type is endtime', () => {
+        default_props.expiry_type = 'endtime';
+        renderAdvancedDuration(mock_store, default_props);
+        expect(screen.getByText(date_picker)).toBeInTheDocument();
+        expect(screen.getByText(time_picker)).toBeInTheDocument();
+    });
+    it('Should render mocked expiry text and should not render trading time picker if contract is not 24 hours and expiry type is endtime', () => {
+        default_props.expiry_type = 'endtime';
+        default_props.duration_units_list = [
+            {
+                text: 'Ticks',
+                value: 't',
+            },
+            {
+                text: 'Seconds',
+                value: 's',
+            },
+            {
+                text: 'Days',
+                value: 'd',
+            },
+        ];
+        renderAdvancedDuration(mock_store, default_props);
+        expect(screen.getByText(expiry_text)).toBeInTheDocument();
+        expect(screen.queryByText(time_picker)).not.toBeInTheDocument();
+    });
+    it('Should render mocked dropdown if duration_units_list length is > 1', () => {
+        renderAdvancedDuration(mock_store, default_props);
+        expect(screen.getByText(dropdown)).toBeInTheDocument();
+    });
+    it('Should not render mocked dropdown if duration_units_list length is > 0', () => {
```

Comment:
Maybe `length` is `0`? Because on the line below you write `default_props.duration_units_list = [];`